cudaaligner/include/claragenomics/cudaaligner/aligner.hpp

```diff
@@ -78,11 +79,25 @@ class Aligner
 /// \param max_target_length Maximum length of target string
 /// \param max_alignments Maximum number of alignments to be performed
 /// \param type Type of aligner to construct
+/// \param allocator Allocator to use for internal device memory allocations
```

Comment:
for the default API, what do you think of hiding the DefaultDeviceAllocator in the implementation itself and not exposing an API with it? e.g. instantiating the DefaultDeviceAllocatin in `allocator.cpp` and passing that into the calls there

Reply:
Sort of like how the other API doesn't expose the allocator itself, and only exposes `max_caching_size`

Reply:
I think the API needs to expose both. I'll add an API call which only exposes `max_caching_size`.